src/mono/msbuild/AppleAppBuilder/Utils.cs

```diff
@@ -0,0 +1,96 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Text;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+internal class Utils
+{
+    public static string GetEmbeddedResource(string file)
+    {
+        using Stream stream = typeof(Utils).Assembly
+            .GetManifestResourceStream("AppleAppBuilder.Templates." + file)!;
+        using var reader = new StreamReader(stream);
+        return reader.ReadToEnd();
+    }
+
+    public static string RunProcess(
+        string path,
+        string args = "",
+        IDictionary<string, string>? envVars = null,
+        string? workingDir = null,
+        bool ignoreErrors = false)
+    {
+        LogInfo($"Running: {path} {args}");
+        var outputBuilder = new StringBuilder();
+        var errorBuilder = new StringBuilder();
+        var processStartInfo = new ProcessStartInfo
+        {
+            FileName = path,
+            UseShellExecute = false,
+            CreateNoWindow = true,
+            RedirectStandardError = true,
+            RedirectStandardOutput = true,
+            Arguments = args,
+        };
+
+        if (workingDir != null)
+        {
+            processStartInfo.WorkingDirectory = workingDir;
+        }
+
+        if (envVars != null)
+        {
+            foreach (KeyValuePair<string, string> envVar in envVars)
+            {
+                processStartInfo.EnvironmentVariables[envVar.Key] = envVar.Value;
+            }
+        }
+
+        Process process = Process.Start(processStartInfo)!;
+        process.ErrorDataReceived += (sender, e) =>
```

Comment:
We should attach ErrorDataReceived and OutputDataReceived before we start the process or we could miss some output.

Reply:
And maybe also follow the recommendations from  https://github.com/dotnet/runtime/issues/18789#issuecomment-252324082

Reply:
> We should attach ErrorDataReceived and OutputDataReceived before we start the process or we could miss some output.

Looks like it won't miss anything, furthermore you can't start listening for outputs before you start the actual process - it throws `StandardOut has not been redirected or the process hasn't started yet`

Reply:
https://github.com/dotnet/runtime/issues/18789#issuecomment-252324082 doesn't apply here since I actually use the parameterless overload